xdsl/dialects/snitch_runtime.py

```diff
@@ -398,14 +403,185 @@ def __init__(self, transfer_id: Operation | SSAValue):
 
 
 @irdl_op_definition
-class DmaWaitAllOp(SnitchRuntimeBarrier):
+class DmaWaitAllOp(NoOperandNoResultBaseOp):
     """
     Block until all operations on the DMA cease
     """
 
     name = "snrt.dma_wait_all"
 
 
+"""
+The number of data movers determines the number of
+independent memory address patterns a core can keep track
+of. Since the data movers are tied to individual registers,
+there need to be at least the same number of registers with
+stream semantics as there are data movers. Multiple SSRs
+may address the same data mover, for example to use the
+data mover both in integer and FP instructions.
+for more info check https://arxiv.org/pdf/1911.08356.pdf
+
+The different SSR data movers.
+    SNRT_SSR_DM0 = 0,
+    SNRT_SSR_DM1 = 1,
+    SNRT_SSR_DM2 = 2,
+"""
+ssr_dm = i32
+
+"""
+The different dimensions - those determine how many levels of nesting a loop can have.
+The snitch system handles cases with up to 4, but this can be extended.
```

Comment:
Thanks for adding this documentation, however this explanation does not sufficiently clear out for me what the operands to these operations mean to me.